catroid/src/main/res/values/strings.xml

```diff
@@ -320,7 +320,7 @@
     <string name="am_convert">Convert</string>
     <string name="dialog_confirm_delete">You can\'t undo this!</string>
 
-    <string name="list_headline_sprites">Objects</string>
+    <string name="list_headline_sprites">Actors and objects</string>
```

Comment:
Just curious about this change. Why `actors`?

Reply:
this changes were proposed by @wslany -> so you'll have to ask him :smirk_cat:

Reply:
@robertpainsi Thank you for asking about the proposed change in terminology, as it gives me the opportunity to explain it here, and also to address some more fundamental questions about terminology used in our project. 

The metaphor we are using in Catrobat, inherited from Scratch, stems from theaters or movies: in theaters or movies as well as in Catrobat, we have a stage with a background, the actors follow their individual scripts individually (objects such as cars etc also move around individually), where the individual pieces of the scripts are mostly formulated as orders to the actor ('Wait 2 seconds ", "Go to random position", ...) or describe changes in objects or backgrounds ("Set mass to 2 kilograms", ...), some of the actors speak and/or ask questions, make sounds, hide and show themselves, go to the back and to the front of the stage, are all initially seen in 2D from the spectators' point of view on the stage, and have several sequential scenes. 

Note that any metaphor never has perfectly identical characteristics as the original, see for instance the "desktop metaphor", but that as a general rule in language use, one should not mix metaphors as much as possible in order to avoid confusion. Metaphors should make it simpler to understand what is meant by words and what can be done with the things and actors, e.g., "folders", or the original "dustbin" on desktop GUIs, where one can take something out of a dustbin in reality as well as in the GUI. That's also why "using good metaphors" is one of the 12 basic practices of the original eXtreme Programming. 

Scratch uses the term "costume" instead of our "look", but look is clearly a closer fit and more understandable in the context of the metaphor because the visible parts of objects and living beings on the stage, e.g., animals or heads of humans or monsters, in theater pieces and movies are obviously better described by the more generic "look" than the much more constrained term "costume". 

Scratch uses the term "Sprite" that, while traditionally having been used among programmers for moving graphical elements in computer games from a developer's point of view, wouldn't be known among our users from any formal programming education, since they wouldn't have had such previous knowledge nor education. "Sprite" also breaks the theater/movie metaphor, as it normally means "elf" or "spirit" in English (in German Kobold, Elfe or Elementargeist), which also does not fit well to the metaphor, e.g., when used for a ball, a human, an alien, car, rocket, or orc monster. The words object and actor are much broader in their usage and thus are a more natural and immediately understandable fit. No "Hmmm, what could that be?" delay... 

Additionally, the name "object" was clearly coming from the programming terminology of our software developer background. It breaks the metaphor unnecessarily: In a theater or movie, besides the background, you simply _do_ have animated/living as well as inanimate entities, in real life *commonly* named "actors" and "objects", which thus make the most sense in our chosen metaphor. Note that they also fit in any translated versions of the words. And they are commonly used and known by our intended target group of teenagers around the world. Moreover, I always thought that using "object" for living beings such as humans somehow felt degrading and insulting, not to mention that ordering them around and letting them speak made little sense for mere inanimate "objects". 

Having longer and more precise while well understandable names and phrases is not only good practice in the Clean Code methodology because it makes a lot of sense for maintainability reasons, it also is already in use everywhere else in our GUI, e.g., when we allow long phrases with spaces and exclamation points or even graphical emojis in variable names, and long phrases in the names of bricks and formula elements. Also, the artificial restriction in most programming languages that a name of a command, variable, message etc needs to be connected by non-space characters, only stems from the need to be able to more easily parse the written code by compilers and interpreters. For human readers, with their more advanced language parsing capabilities, this is not necessary and actually degrades readability. 

In the same context, I also strongly believe that "When scene starts" is a better name than "When project starts" for that brick: Not only does it perfectly fit the metaphor, even in theater pieces or movies with a _single_ scene, it also emphasizes the new concept of scenes we have introduced. It thereby pushes the concept of our scenes to the forefront of awareness among our users, which is a good thing as scenes are a bit difficult to discover as of now. Nevertheless, we still want to keep scenes a bit more in the background for simpler projects with single scenes, just hinting at "something more" through this new name of the "When scene starts" of the old "When project starts" brick. Scenes can thus be discovered by more advanced users, but with a clear meaning nevertheless even for users that do not get any additional explanation about scenes. Intuitiveness at its best ;-). It also captures much better the intended meaning for any _later_ scenes than the currently "When project starts" brick, which is currently named incorrectly in these second, third, etc scenes.

So, altogether, actors and objects are better suited in Catrobat than the term "object" alone. Since I wanted to correct many other things in strings.xml anyway as well, and considering the need to correct many translations anyway because of all these changes, I thought that now was a good opportunity to correct this unfortunate naming inconsistency as well. The change of name from "program" to "project" goes along the same lines. 

If you spot any other formulations that should be corrected now, please suggest them --- I'm almost sure I didn't catch them all. Consider it a proposal for discussion instead of an already decided change request. However, this format of a pull request allows to easily see all changes precisely, allows to suggest alternatives, and allows to easily discuss them one by one.

Reply:
Thanks a lot for your detailed explanation and documentation.

Any concerns that users will think that actors and objects are somehow different (different functionality, behavior)? → needing to explain that functionally and behaviorally speaking, they are not.

Reply:
Hmm, I don't think that there need to be any concerns. Using both terms together as proposed is more inclusive, because of the limitations of the natural language and what users know. For computer science people with a background in object oriented programming, it is clear that a person can be an object, but that is not so in real life, and our users only have a background in real life, but not a formal education in computer science or object oriented programming. If there would be a good phrase for both actors and objects (which unfortunately "sprite" isn't), then we could use it instead. But there seems to be none. And the analogy from reality is quite clear. One aspect that may possibly be done differently at some time later (if ever) would be to have different categories in the media gallery, or to show them in different ways. But I don't think that it would warrant handling them in different ways. In many cases, it might be OK to have "objects" that are somehow personified, e.g., a "car" that at the same time is somehow "alive" and has its own "personality", like in the movie "Cars", can be seen both as an actor than as an "object". Also, since the order in the list of actors and objects implies the default layer, it is important to be able to mix actors and objects in the list, so they cannot be kept separate in different lists.